google/web-stories-wp
# Layers Panel: Duplicate layer from panel

## Context
Part of updates to the layers panel



## Summary
Adds duplicate action to the layers in the layers panel:
<img width="334" alt="Screen Shot 2021-10-15 at 2 39 49 PM" src="https://user-images.githubusercontent.com/35983235/137544532-85030624-833a-4d13-bf9b-1d774a759d83.png">


## Relevant Technical Choices
Created a new story reducer action for this.




## To-do
- [ ] Discuss UX. It's a little hard to tell which layer was the original and which was the duplicate once you duplicate:
https://user-images.githubusercontent.com/35983235/137544933-8e23818f-650b-4af4-92d6-7165f4b6e322.mp4


## User-facing changes
Layers now have a `duplicate` action on them
<img width="334" alt="Screen Shot 2021-10-15 at 2 39 49 PM" src="https://user-images.githubusercontent.com/35983235/137544532-85030624-833a-4d13-bf9b-1d774a759d83.png">


## Testing Instructions





This PR can be tested by following these steps:

1. Open a story
2. Make sure there are some elements in it
3. go to the layers panel
4. hover over any layer and see the duplicate layer button
5. see that the element duplicates when you press it


## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #9137
